### ISSUE TITLE
Add Search Field

### ISSUE THREAD

**Issue by JPluess** [Gerrnperl/outline-map#47]
Hey, I would really appreciate having a search bar. Additionally, it would be great to have a button that toggles between searching for variable definitions, functions, and classes. This feature would significantly speed up my workflow

**Comment by Gerrnperl** [comment]
Thank you for your very inspiring idea. I have already implemented this.

![Preview](https://github.com/Gerrnperl/outline-map/assets/64595686/6924faac-faf7-4e1c-ad18-37feb463fb7d)

I will release this version after some minor modifications.

**Comment by JPluess** [comment]
excellent, thanks for improving the tool, is it possible to have the searchfield extend to fit the panel width, 
![grafik](https://github.com/Gerrnperl/outline-map/assets/7995240/3adb8d99-a2f6-410d-aa23-de9b8101207d)

**Comment by Gerrnperl** [comment]
I have fixed the style of search field.

---

For the second, it should not automatically hide after triggering visible(alt+l).
Do you want it to be visible when the view is launched?

**Comment by JPluess** [comment]
I think there is enough space to have it always visible, no need for pinning.

**Comment by Gerrnperl** [comment]
Considering some users just place the outline in a sub view, I think the ability to hide the search bar should be retained. Now I modified it to stay in the state when the editor is closed and not hide when using the shortcut key to focus the search. The effect should be the same for users who don't want it to be hidden.

**Comment by JPluess** [comment]
You're absolutely right; preserving the settings is a more convenient approach that can cater to everyone's needs. Thank you for bringing this important aspect to my attention.

I usually navigate from the Explorer view. Would it be possible to integrate the full map feature with search functionality into this section?  There is a horizontal tab called "Gliederung" exploring files and filecontent seems an appropriate place, the standalone section as is could still be there for those who like it that way.  Here is an image to display my idea

![grafik](https://github.com/Gerrnperl/outline-map/assets/7995240/342c5672-779e-4e4b-919c-38f829c9697a)

**Comment by Gerrnperl** [comment]
VSCode supports moving any view to any panel. Thus, you can just drag this tab to the place you want.
Here is an example.
![example](https://github.com/Gerrnperl/outline-map/assets/64595686/e4bffaa5-b2c2-4b70-b1bc-de3e656d0200)

**Comment by JPluess** [comment]
Thank you so much; I learned something new. Regarding the search functionality:
1.     Is it possible to search across all open files?
2.     Can we also search any file in the current project?
Thanks a lot.

**Comment by Gerrnperl** [comment]
vscode provides a method to search for symbols in the workspace. You can use Ctrl-T to search for symbols in the workspace (you can also use Ctrl-Shift-O to search for symbols in the current file). vscode handles symbols in the workspace and in a single file differently. I cannot directly obtain all the symbols in the workspace. I can only "search" for the workspace symbols. I think it is not necessary to implement this built-in function of vscode again.

**Comment by Gerrnperl** [comment]
You'd better open a new issue and provide some information for reproducing...
1. It may occur after an interrupted search, and the search runs into a unnormal mode.
2. The map may not load if vscode doesn't return any symbol information. I have no idea how to resolve it.
3. I have not encountered this situation, please provide the information needed to reproduce it

**Comment by JPluess** [comment]
An extended idea for the search functionality could involve adding preset filters such as

1.  "Expand/Contract" 
2. "Show/Hide" for variables, functions, classes, and other elements. 

This would help declutter the panel and improve usability. You could implement small toggle buttons for filtering
